cypress/support/commands.ts

```diff
@@ -0,0 +1,253 @@
+declare global {
+  // eslint-disable-next-line @typescript-eslint/no-namespace
+  namespace Cypress {
+    interface Chainable {
+      getByTestId(
+        selector: string,
+        ...additionalArguments: any[]
+      ): Chainable<Element>;
+
+      getDropIndicator(
+        indicator: 'error' | 'success',
+        indicatorColors?: Record<'error' | 'success', string>
+      ): Chainable<Element>;
+
+      dragAndDrop(
+        target: string,
+        options?: Partial<DropOptions>
+      ): Chainable<Element>;
+
+      dragOver(
+        target: string,
+        options?: Partial<DropOptions>
+      ): Chainable<Element>;
+
+      drop(): Chainable<Element>;
+    }
+  }
+}
+
+export interface DropOptions {
+  position: 'right' | 'left' | 'right' | 'above' | 'below' | 'inside';
+}
+
+/**
+ * Will select elements that have a data-cy=$test-id attribute
+ * Usage: `cy.getByTestId('test-id')`
+ * @example HTML: <div data-cy="test-id" /> Selector: cy.getByTestId('test-id')
+ */
+Cypress.Commands.add('getByTestId', (selector, ...args) => {
+  return cy.get(`[data-cy="${selector}"]`, ...args);
+});
+
+/**
+ * Get either the 'success' or 'error' drop indicator
+ * Use the indicatorColors parameter when you have overwritten the indicator colors
+ * @example cy.getDropIndicator('success')
+ */
+Cypress.Commands.add(
+  'getDropIndicator',
+  (
+    indicator: 'error' | 'success',
+    indicatorColors: Record<'error' | 'success', string> = {
+      error: 'red',
+      success: 'rgb(98, 196, 98)',
+    }
+  ) => {
+    // in some cases rendering the indicator is slower than cypress, so we have to add a short wait
+    cy.wait(250);
+
+    return cy.get('div').filter((index, element) => {
+      return (
+        // the next.js render indicator shares some of the styles but additionally has an id
+        element.id === '' &&
+        element.classList.length === 0 &&
+        element.style.position === 'fixed' &&
+        element.style.zIndex === '99999' &&
+        element.style.backgroundColor === indicatorColors[indicator]
+      );
+    });
+  }
```

Comment:
Can we move all the dnd-related commands (ie: DummyDataTransfer, getDropIndicator, drop etc) into a separate file; maybe `dnd.ts` and then just import that here?

Reply:
Done. In phase 2 I want to publish the actions in `dnd.ts` as a separate `@craft/cypress` package :)